diff --git a/.gitconfig b/.gitconfig
new file mode 100644
--- /dev/null
+++ ./.gitconfig
@@ -0,0 +1,16 @@
+[remote "origin"]
+    url = git@github.com:huangyingw/oh-my-zsh.git
+    fetch = +refs/heads/*:refs/remotes/origin/*
+[remote "robbyrussell"]
+	url = https://github.com/robbyrussell/oh-my-zsh.git
+	fetch = +refs/heads/master:refs/remotes/robbyrussell/master
+[gsync]
+    remote = robbyrussell
+    target = robbyrussell/master
+[push]
+    remote = origin
+[branch "dev"]
+	remote = origin
+	merge = refs/heads/dev
+[gdif]
+    ignore = dev.gdio.diff
diff --git a/.gitmodules b/.gitmodules
new file mode 100644
--- /dev/null
+++ ./.gitmodules
@@ -0,0 +1,3 @@
+[submodule "custom/plugins/zsh-nvm"]
+	path = custom/plugins/zsh-nvm
+	url = https://github.com/lukechilds/zsh-nvm
diff --git a/custom/plugins/zsh-nvm b/custom/plugins/zsh-nvm
new file mode 160000
--- /dev/null
+++ ./custom/plugins/zsh-nvm
@@ -0,0 +1 @@
+Subproject commit 63595c2579daa79d840499d63d41e4648fccfa1c
diff --git a/lib/history.zsh b/lib/history.zsh
--- ./lib/history.zsh
+++ ./lib/history.zsh
@@ -34,7 +34,6 @@ esac
 ## History command configuration
 setopt extended_history       # record timestamp of command in HISTFILE
 setopt hist_expire_dups_first # delete duplicates first when HISTFILE size exceeds HISTSIZE
-setopt hist_ignore_dups       # ignore duplicated commands history list
 setopt hist_ignore_space      # ignore commands that start with space
 setopt hist_verify            # show command with history expansion to user before running it
 setopt share_history          # share command history data
