diff --git a/.gitmodules b/.gitmodules
new file mode 100644
index 00000000..102ce18f
--- /dev/null
+++ ./.gitmodules
@@ -0,0 +1,3 @@
+[submodule "custom/plugins/zsh-nvm"]
+	path = custom/plugins/zsh-nvm
+	url = https://github.com/lukechilds/zsh-nvm
diff --git a/custom/plugins/zsh-nvm b/custom/plugins/zsh-nvm
new file mode 160000
index 00000000..23067bd9
--- /dev/null
+++ ./custom/plugins/zsh-nvm
@@ -0,0 +1 @@
+Subproject commit 23067bd9bb6eb6f4737a3ea90cb0cb5e85f61ba2
diff --git a/lib/history.zsh b/lib/history.zsh
index 79407690..deaa4cc3 100644
--- ./lib/history.zsh
+++ ./lib/history.zsh
@@ -34,7 +34,6 @@ esac
 ## History command configuration
 setopt extended_history       # record timestamp of command in HISTFILE
 setopt hist_expire_dups_first # delete duplicates first when HISTFILE size exceeds HISTSIZE
-setopt hist_ignore_dups       # ignore duplicated commands history list
 setopt hist_ignore_space      # ignore commands that start with space
 setopt hist_verify            # show command with history expansion to user before running it
 setopt share_history          # share command history data
